client/src/App.jsx

```diff
@@ -13,18 +13,25 @@ const withSuspense = (Component) => (
 
 const HomePage = React.lazy(() => import("./pages/HomePage"));
 const SearchResultsPage = React.lazy(() => import("./pages/SearchResultsPage"));
+const AboutPage = React.lazy(() => import("./pages/AboutPage"));
+
+export const AuthContext = createContext([{}, () => {}]);
 
 const App = () => {
+  const [authUser, setAuthUser] = useState(null);
   return (
-    <BrowserRouter>
-      <CssBaseline />
-      <ThemeProvider theme={theme}>
-        <Routes>
-          <Route path={"/"} element={withSuspense(HomePage)} />
-          <Route path={"/search"} element={withSuspense(SearchResultsPage)} />
-        </Routes>
-      </ThemeProvider>
-    </BrowserRouter>
+    <AuthContext.Provider value={[authUser, setAuthUser]}>
+      <BrowserRouter>
+        <CssBaseline />
+        <ThemeProvider theme={theme}>
+          <Routes>
+            <Route path={"/"} element={withSuspense(HomePage)} />
+            <Route path={"/search"} element={withSuspense(SearchResultsPage)} />
+            <Route path={"/about"} element={withSuspense(AboutPage)} />
```

Comment:
Can we take a look at this? i don't think we're going to benefit from suspense and code splitting here. Clicking between the pages we're hitting the suspense, but React is not caching the already loaded assets. There's a chance the `withSuspense` pattern we established here isn't going to be the best approach (if this assumption of cause is correct).

![2023-04-30 22 25 33](https://user-images.githubusercontent.com/12107969/235410999-e278bc68-0f54-47e7-b0db-07d8a1a04d0b.gif)

Reply:
I'm not very familiar with the concept. Do you know where i can learn more about this? Do you mean like the navbar (which is the same across all the pages) should be cached and not re-rendered?

Reply:
https://github.com/JennaXiao3/internship-tracker/pull/1

Heres an example of the intentional usage of a Link from react router vs Link from MUI. MUI provides the _styling_, but react router provides the _functionality_ when it comes to client side routing

see [react router](https://reactrouter.com/en/main/start/concepts) for some of the core concepts

Reply:
https://deploy-preview-114--bt-internship-tracker.netlify.app/

Here's a preview with the changes i made, you'll see the initial flash becuase we have an empty page with the "brewing..." text, and a white background, but once we've loaded those, react is now handling the quick hot swap of only the components that need to swap out as the router controls the rendering cycles

Reply:
ohhh that's so coool

Reply:
Nice, did you see the difference?

Reply:
YAHHHH thank youuu I've never paid attention to that before but definitely good to know for the future :)